### ISSUE TITLE
WinUI3 Preview 3 package misses globbing pattern for PRIResource

### ISSUE THREAD

**Issue by huoyaoyuan** [microsoft/microsoft-ui-xaml#3648]
**Issues I've met when upgrading a functioning project from preview2 to preview3:**

### Can't find ResourceMap:
https://github.com/microsoft/microsoft-ui-xaml/issues/3620#issuecomment-729589622
/cc @roxk

#### Cause:
The preview 3 nuget package doesn't include a glob pattern for `PRIResource`, while preview 2.

#### Workaround:
Add this to your project file:
```xml

  <ItemGroup>
    <PRIResource Include="Strings\**\*.resw" />
    <!-- Strings may be changed to other folder you use, or just **\*.resw -->
  </ItemGroup>

```

<s>
### XamlParseException

```
onecoreuap\base\appmodel\statemanager\winrt\lib\windows.storage.applicationdatafactory.server.cpp(235)\Windows.Storage.ApplicationData.dll!00007FFD72EF83F3: (caller: 00007FFD72F087B3) ReturnHr(1) tid(695c) 8000000B 该操作尝试访问的数据超出了有效范围
```

Wrapped by `XamlParseException`.

Currently no idea. Investigating.
</s>

Edit: this one should be unrelated.

**Comment by roxk** [comment]
@huoyaoyuan Thanks for the heads up. In my case, I found that the `.resw` files are already included in the project file so my crash seems to be different than yours. Specifically, `XXX.Label` and `XXX.Text` works fine, but `XXX.ToolTipService.ToolTip` would crash. Workaround is just removing these tool tip resources😅 I will file a new issue once I have a minimal repro.

**Comment by stevenbrix** [comment]
Interesting, I'm not aware of anything that would have changed here. @rohanp-msft, do you know if this is due to using the new PRI gen tooling?

**Comment by rohanp-msft** [comment]
> Interesting, I'm not aware of anything that would have changed here. @rohanp-msft, do you know if this is due to using the new PRI gen tooling?

looking now

**Comment by rohanp-msft** [comment]
@huoyaoyuan Hi Huo, what kind of winui3 project is this? Examples are below:
![image](https://user-images.githubusercontent.com/35987549/101421743-e5257d80-38a9-11eb-9a74-d56f53f0e8e3.png)

**Comment by huoyaoyuan** [comment]
@rohanp-msft C# desktop, packaged. The project was manually updated from preview 2, which didn't request adding entry.

**Comment by rohanp-msft** [comment]
@stevenbrix For C# desktop, packaged apps, no, the new PRI gen tooling didn't introduce the need to do this. (The reason I asked about the type of project is that for class lib projects, I noticed that, both with winui 2 and 3, string resources don't make it into the class lib's PRI. That's what I'm working on figuring out now.)
CC @huoyaoyuan

**Comment by huoyaoyuan** [comment]
@rohanp-msft I mistyped it. I was using C# classlib.

**Comment by rohanp-msft** [comment]
@huoyaoyuan Huo, that's something I've seen before. I tried to repro it today but couldn't. Here's the issue where I recorded my repro attempt: https://github.com/microsoft/ProjectReunion/issues/246
Can you repro it with a new WinUI class lib project? If yes, can you share your setup like what I shared in the link, please?

**Comment by huoyaoyuan** [comment]
@rohanp-msft Adding a resw using Visual Studio dialog adds `<PRIResource Include="Resources.resw" />` in the project file. If it's commented out, Visual Studio shows it as `None`.

**Comment by rohanp-msft** [comment]
@huoyaoyuan That's good to know. Was that line present in the project file before you upgraded to WinUI 3.0 from WinUI 2.0? Did it go away after the upgrade?

**Comment by huoyaoyuan** [comment]
@rohanp-msft  I upgraded from WinUI 3 Preview 2. Preview 2 didn't require the line. I did the upgrade manually.

**Comment by rohanp-msft** [comment]
OK, thanks for the info, @huoyaoyuan. I'll try to repro that (creating a new WinUI 3 Preview 3 project works for me, but if I try upgrading to it, perhaps it won't).

**Comment by rohanp-msft** [comment]
@huoyaoyuan Would you please share the pre-migration project file? Or point me to it?

**Comment by sjb-sjb** [comment]
I had a similar problem. In addition to adding the globbing pattern in a PRIResource tag, I had to go to each of the various .resw files in my project and verify that the build action was set to PRIResource. Even though all of the .resw files were added to the project in the same manner (by copying them into the project directory using FIle Explorer) some of them had the PRIResource build action while others did not. 

Comment: at first, I accidentally (and incorrectlyi) used a path such as Include="\**\Library\**\*.resw". This caused massive problems including for example extremely long time to just Clean the solution.

**Comment by JeanRoca** [comment]
@huoyaoyuan this should be fixed by preview 4.

**Comment by huoyaoyuan** [comment]
Preview 4 behaves the same with Preview 3. Waiting for it.

**Comment by huoyaoyuan** [comment]
@JeanRoca It's still missing in Reunion 0.5.

**Comment by rohanp-msft** [comment]
> @JeanRoca It's still missing in Reunion 0.5.

Hi @huoyaoyuan,
As you pointed out, this falls under https://github.com/microsoft/ProjectReunion/issues/613.
You beat me to suggesting duping this to that bug.
Sorry about the delay here. The issue was not fully understood until recently.
The fix should be available in v0.8 preview of Project Reunion which starts shipping this month.
Please follow https://github.com/microsoft/ProjectReunion/issues/613 to track completion.
Feel free to dupe this issue to https://github.com/microsoft/ProjectReunion/issues/613.

**Comment by huoyaoyuan** [comment]
Solved in 0.8 stable.
